tests/aggregators.rs

```diff
@@ -0,0 +1,769 @@
+            .with_api_headers()
+            .with_state(user)
+            .with_request_json(json!({
+                "role": "whatever",
+                "bearer_token": "not valid base64",
+                "api_url": "not a url",
+                "dap_url": "also not a url"
+            }))
+            .run_async(&app)
+            .await;
+
+        assert_response!(conn, 400);
+        let error: Value = conn.response_json().await;
+        assert!(error.get("role").is_some());
+        assert!(error.get("name").is_some());
+        assert!(error.get("api_url").is_some());
+        assert!(error.get("bearer_token").is_some());
+        assert!(error.get("dap_url").is_some());
+        Ok(())
+    }
+
+    #[test(harness = set_up)]
+    async fn not_member(app: DivviupApi) -> TestResult {
+        let user = fixtures::user();
+        let account = fixtures::account(&app).await; // no membership
+
+        let aggregator_count_before = Aggregators::find().count(app.db()).await?;
+        let mut conn = post(format!("/api/accounts/{}/aggregators", account.id))
+            .with_api_headers()
+            .with_state(user)
+            .with_request_json(fixtures::new_aggregator())
+            .run_async(&app)
+            .await;
+
+        assert_not_found!(conn);
+        let aggregator_count_after = Aggregators::find().count(app.db()).await?;
+        assert_eq!(aggregator_count_before, aggregator_count_after);
+
+        Ok(())
+    }
+
+    #[test(harness = set_up)]
+    async fn nonexistant_account(app: DivviupApi) -> TestResult {
+        let user = fixtures::user();
+        let aggregator_count_before = Aggregators::find().count(app.db()).await?;
+
+        let mut conn = post("/api/accounts/does-not-exist/aggregators")
+            .with_api_headers()
+            .with_state(user)
+            .with_request_json(fixtures::new_aggregator())
+            .run_async(&app)
+            .await;
+
+        assert_not_found!(conn);
+        let aggregator_count_after = Aggregators::find().count(app.db()).await?;
+        assert_eq!(aggregator_count_before, aggregator_count_after);
+
+        Ok(())
+    }
+
+    #[test(harness = set_up)]
+    async fn admin_not_member(app: DivviupApi) -> TestResult {
+        let (admin, ..) = fixtures::admin(&app).await;
+        let account = fixtures::account(&app).await;
+        let mut conn = post(format!("/api/accounts/{}/aggregators", account.id))
+            .with_api_headers()
+            .with_state(admin)
+            .with_request_json(fixtures::new_aggregator())
+            .run_async(&app)
+            .await;
+
+        assert_response!(conn, 201);
+        let aggregator: Aggregator = conn.response_json().await;
+        assert_eq!(
+            Aggregators::find_by_id(aggregator.id)
+                .one(app.db())
+                .await?
+                .unwrap(),
+            aggregator
+        );
+        Ok(())
+    }
```

Comment:
Are there tests for admins creating shared aggregators and for members not being able to create shared aggregators?

Reply:
There were not! Thanks for catching that. I've added them in 5102c28cf751e684d46f04db467ae011d8d76999